transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollDatagramChannel.java

```diff
@@ -538,16 +538,26 @@ void epollInReady() {
         }
     }
 
-    private boolean connectedRead(EpollRecvByteAllocatorHandle allocHandle, ByteBuf byteBuf)
+    private boolean connectedRead(EpollRecvByteAllocatorHandle allocHandle, ByteBuf byteBuf, int maxDatagramPacketSize)
             throws Exception {
         try {
-            allocHandle.attemptedBytesRead(byteBuf.writableBytes());
-            allocHandle.lastBytesRead(doReadBytes(byteBuf));
+            int writable = maxDatagramPacketSize != 0 ? Math.min(byteBuf.writableBytes(), maxDatagramPacketSize)
+                    : byteBuf.writableBytes();
+            allocHandle.attemptedBytesRead(writable);
+
+            byteBuf.writerIndex(byteBuf.readerIndex() + writable);
```

Comment:
I'm not sure this is right.. isn't it the capacity that would need to be reduced in the case that `0 < maxDatagramPacketSize < byteBuf.writeableBytes()`? Though i'm not sure we would really want to do that since it could involve actually reallocation/copying internally...

Reply:
@njhil arg... I think you are right and it should be: `byteBuf.writerIndex(byteBuf.capacity() - writable);`

Reply:
also I think we should update the readerIndex to match this.

Reply:
actually also this is not 100 % correct... Should be good now with the latest change. PTAL again